### ISSUE TITLE
USE server DNS

### ISSUE THREAD

**Issue by LiveChief** [shadowsocks/shadowsocks-android#2026]
Can we remove the DNS settings and instead just use the normal DNS from server Json file and if that DNS isn't there let's use the servers DNS...

Just the way it's setup on liev on desktop

**Comment by kimw** [comment]
See https://discourse.shadowsocks.org/t/should-the-remote-dns-option-be-removed/269

**Comment by kimw** [comment]
I haven't read the Android code yet. But I've just noticed, that the DNS settings is not working in Android. I mean, whatever I set it, the mobile is just querying DNS from the DNS server set on ss-server side.

BTW, this behaviour brings some interesting feature, e.g. I can block website from ss-server side by:

1. Set up a DNS server (e.g. dnscrypt-proxy which is able to block DNS by simply put them into blocked.txt file) listen on VPS's localhost (for avoid public service).
2. Set up ss-server to query DNS from this localhosted DNS server.

This one-set-up is able to working for ALL `ss-local`s. Emm, it's cool (to block the big brother's companies).
